unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -2597,6 +2598,79 @@ bool includeInvisible
             actionFinished(true);
         }
 
+        public void Get2DSemanticHulls(List<string> objectTypes = null) {
+            HashSet<string> allowedObjectTypesSet = null;
+            if (objectTypes != null) {
+                allowedObjectTypesSet = new HashSet<string>(objectTypes);
+            }
+
+            var objectIdToConvexHull = new Dictionary<string, List<(float x, float y)>>();
+            foreach (var sop in GameObject.FindObjectsOfType<SimObjPhysics>()) {
+                if (
+                    allowedObjectTypesSet != null 
+                    && !allowedObjectTypesSet.Contains(sop.Type.ToString())
+                ) {
+                    continue;
+                }
+                var vertices = new List<MIConvexHull.DefaultVertex2D>();
+                float maxY = -float.PositiveInfinity;
+
+                foreach (var meshFilter in sop.GetComponentsInChildren<MeshFilter>()) {
+                    foreach (var localVertex in meshFilter.mesh.vertices) {
+                        var globalVertex = meshFilter.transform.TransformPoint(localVertex);
+                        vertices.Add(new MIConvexHull.DefaultVertex2D(globalVertex.x, globalVertex.z));
+                        maxY = Math.Max(maxY, globalVertex.y);
+                    }
+                }
+                var miconvexHull = MIConvexHull.ConvexHull.Create2D(vertices, 1e-10);
+                objectIdToConvexHull[sop.ObjectID] = miconvexHull.Result.Select(p => ((float) p.X, (float) p.Y)).ToList();
+
+                #if UNITY_EDITOR
+                Debug.Log(sop.ObjectID);
+                var pointsOnHullArray = miconvexHull.Result.ToArray();
+                for (int i = 0; i < pointsOnHullArray.Length; i++) {
+                    var p0 = pointsOnHullArray[i];
+                    var p1 = pointsOnHullArray[(i + 1) % pointsOnHullArray.Length];
+                    Debug.DrawLine(
+                        new Vector3((float) p0.X, maxY, (float) p0.Y),
+                        new Vector3((float) p1.X, maxY, (float) p1.Y),
+                        Color.red, 100.0f
+                    );
+                }
+                #endif
+            }
+            actionFinishedEmit(true, objectIdToConvexHull);
+        }
+        
+        public void GetPixelToWorldPoint(float row, float col, float depth) {
+            actionFinishedEmit(
+                true, 
+                m_Camera.ScreenToWorldPoint(
+                    new Vector3(col, m_Camera.pixelHeight - row, m_Camera.nearClipPlane)
+                )
+            );
+        }
+
+        public void CameraRayCast(float x, float y) {            
+            Ray ray = m_Camera.ViewportPointToRay(new Vector3(x, y, 0.0f));
```

Comment:
are you sure `y` shouldn't be `1-y`, like it is in `getTargetObject`, which ensures that `y=0` is at the top and `y=1` is at the bottom?:
https://github.com/allenai/ai2thor/blob/master/unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs#L4637

Reply:
I've decided to remove this and the `GetPixelToWorldPoint` functions. They're useful for debugging but perhaps for now we don't want them in the public api.